src/frontend/src/utils/iiConnection.ts

```diff
@@ -643,11 +643,13 @@ export class Connection {
     if (this.overrideActor !== undefined) {
       return this.overrideActor;
     }
+
+    const shouldFetchRootKey =
+      features.FETCH_ROOT_KEY || Boolean(this.canisterConfig.fetch_root_key[0]);
```

Comment:
Using boolean constructor on a `boolean | undefined` value is a bit confusing. The following would more explicitly indicate that false is the default:

```suggestion
      features.FETCH_ROOT_KEY || (this.canisterConfig.fetch_root_key[0] ?? false);
```